R/prior.R

```diff
@@ -0,0 +1,56 @@
+#' Define prior from defaults, model, family, user
+#'
+#' @rdname epidist_prior
+#' @family prior
+#' @export
+epidist_prior <- function(data, family, formula, prior) {
+  epidist_validate(data)
+  default <- brms::default_prior(formula, data = data)
+  model <- epidist_model_prior(data, formula)
+  family <-  epidist_family_prior(family, formula)
+  prior <- Reduce(replace_prior, list(default, model, family, prior))
+  return(prior)
+}
+
+#' Model specific prior
+#'
+#' @rdname epidist_model_prior
+#' @family prior
+#' @export
+epidist_model_prior <- function(data, ...) {
+  UseMethod("epidist_model_prior")
+}
+
+#' @family prior
+#' @export
+epidist_model_prior.default <- function(data, formula) {
+  # Currently there are not model-specific priors
+  # In future there might be, but we need to be careful about Stan code
```

Comment:
This is not going to be a good design I think..

Reply:
Can make issue about it or write something about it in existing issue

Reply:
can you touch more on the problems you see?

Reply:
https://github.com/epinowcast/epidist/issues/199#issuecomment-2293080355